 # **Project Name** - **Booking.com - Hotel Booking Analysis**

##### **Project Type**    - EDA
##### **Contribution**    - Individual

# **Problem Statement**

##### Booking.com manages a wide range of hotel bookings across various locations, customer profiles, and channels. The dataset includes key details like booking lead times, room types, customer preferences, and reservation statuses. The challenge is to analyze how factors like booking windows, guest demographics, and reservation types affect hotel performance and customer experience. By leveraging these insights, Booking.com aims to **optimize booking efficiency, predict guest needs, and improve customer satisfaction, ensuring a competitive edge in the dynamic hospitality industry**

# **General Guidelines :**

#***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### Dataset Loading

In [188]:
df = pd.read_csv('/content/drive/MyDrive/Local Disk (D:)/Data Analyst/Project_Root/Data/Raw/Hotel Bookings.csv')

### Dataset First View

In [4]:
df.head()

hotel  is_canceled  lead_time  arrival_date_year arrival_date_month  \
0  Resort Hotel            0        342               2015               July   
1  Resort Hotel            0        737               2015               July   
2  Resort Hotel            0          7               2015               July   
3  Resort Hotel            0         13               2015               July   
4  Resort Hotel            0         14               2015               July   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        27                          1   
1                        27                          1   
2                        27                          1   
3                        27                          1   
4                        27                          1   

   stays_in_weekend_nights  stays_in_week_nights  adults  ...  deposit_type  \
0                        0                     0       2  ...    No Deposit   
1                        0                     0       2  ...    No Deposit   
2                        0                     1       1  ...    No Deposit   
3                        0                     1       1  ...    No Deposit   
4                        0                     2       2  ...    No Deposit   

   agent company days_in_waiting_list customer_type   adr  \
0    NaN     NaN                    0     Transient   0.0   
1    NaN     NaN                    0     Transient   0.0   
2    NaN     NaN                    0     Transient  75.0   
3  304.0     NaN                    0     Transient  75.0   
4  240.0     NaN                    0     Transient  98.0   

   required_car_parking_spaces  total_of_special_requests  reservation_status  \
0                            0                          0           Check-Out   
1                            0                          0           Check-Out   
2                            0                          0           Check-Out   
3                            0                          0           Check-Out   
4                            0                          1           Check-Out   

  reservation_status_date  
0              2015-07-01  
1              2015-07-01  
2              2015-07-02  
3              2015-07-02  
4              2015-07-03  

[5 rows x 32 columns]

### Dataset Rows & Columns count

In [84]:
df.shape

(119390, 32)

### Dataset Information

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 32 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  object 
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  object 
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_week_nights            119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal            

In [77]:
null_values = df.isnull().sum()
columns_with_nulls = null_values[null_values > 0]
print(columns_with_nulls)

children         4
country        488
agent        16340
company     112593
dtype: int64


### What did you know about your dataset?

In [209]:
!git status
!git add .
!git commit -m "Dataset loaded"

On branch master
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	new file:   Data/Raw/Hotel Bookings.csv
	modified:   Notebooks/Booking.com - Hotel Booking Analysis.ipynb

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Notebooks/Booking.com - Hotel Booking Analysis.ipynb

[master 854c8fa] Dataset loaded
 2 files changed, 119392 insertions(+), 1 deletion(-)
 create mode 100644 Data/Raw/Hotel Bookings.csv
 rewrite Notebooks/Booking.com - Hotel Booking Analysis.ipynb (81%)


##### Booking offers two types of hostels across 177 countries. The dataset contains 119390 rows and 32 columns. Four columns—children, country, agent, and company—have missing/NaN values, with the company column having the highest number of NaN values.

## ***2. Understanding Your Variables***

In [87]:
df.columns

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type', 'agent',
       'company', 'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date'],
      dtype='object')

#### Variables Description

* **hotel** : hotel type (City Hotel, Resort Hotel)
* **is_canceled :** Booking canceled(1) and if not canceled=(0)
* **lead_time :** Elapsed days between entering date of booking and arrival date
* **arrival_date_year :** Year of arrival
* **arrival_date_month :** Month of arrival
* **arrival_date_week_number :** Week number of arrival date
* **arrival_date_day_of_month :** Date of arrival
* **stays_in_weekend_nights :** Number of weekend night (Saturday or Sunday) the guest stayed or booked in the hotel
* **stays_in_week_nights :** Number of weekday night (Monday to Friday) the guest stayed or booked in the hotel
* **adults :** Number of adults
* **children :** Number of children
* **babies :** Number of babies
* **meal :** Kind of mean
* **country :** Country code
* **market_segment :** Which segment customer belong to
* **distribution_channel :** How customer assessed stay - corporate booking/Direct/TA.TO
* **is_repeated_guest :** If guest coming first time(0) else (1)
* **previous_cancellations :** Was there cancelation before
* **previous_bookings_not_canceled :** Previous bookings was not canceled
* **reserved_room_type :** Types of room reserved
* **assigned_room_type :** Type of room assined to guest
* **booking_changes :** If there was any change made in booking
* **deposit_type :** Type of deposit - ['No Deposit', 'Refundable', 'Non Refund']
* **agent :** Booking through agent
* **company :** Company Name
* **days_in_waiting_list :** Number of days in waitlist
* **customer_type :** Type of customer
* **adr :** Average daily rate
* **required_car_parking_spaces :** If car parking is required
* **total_of_special_requests :** Number of special requirement
* **reservation_status :** Reservation status
* **reservation_status_date :** Reservation status date

In [92]:
df.describe(include='all')

hotel    is_canceled      lead_time  arrival_date_year  \
count       119390  119390.000000  119390.000000      119390.000000   
unique           2            NaN            NaN                NaN   
top     City Hotel            NaN            NaN                NaN   
freq         79330            NaN            NaN                NaN   
mean           NaN       0.370416     104.011416        2016.156554   
std            NaN       0.482918     106.863097           0.707476   
min            NaN       0.000000       0.000000        2015.000000   
25%            NaN       0.000000      18.000000        2016.000000   
50%            NaN       0.000000      69.000000        2016.000000   
75%            NaN       1.000000     160.000000        2017.000000   
max            NaN       1.000000     737.000000        2017.000000   

       arrival_date_month  arrival_date_week_number  \
count              119390             119390.000000   
unique                 12                       NaN   
top                August                       NaN   
freq                13877                       NaN   
mean                  NaN                 27.165173   
std                   NaN                 13.605138   
min                   NaN                  1.000000   
25%                   NaN                 16.000000   
50%                   NaN                 28.000000   
75%                   NaN                 38.000000   
max                   NaN                 53.000000   

        arrival_date_day_of_month  stays_in_weekend_nights  \
count               119390.000000            119390.000000   
unique                        NaN                      NaN   
top                           NaN                      NaN   
freq                          NaN                      NaN   
mean                    15.798241                 0.927599   
std                      8.780829                 0.998613   
min                      1.000000                 0.000000   
25%                      8.000000                 0.000000   
50%                     16.000000                 1.000000   
75%                     23.000000                 2.000000   
max                     31.000000                19.000000   

        stays_in_week_nights         adults  ...  deposit_type          agent  \
count          119390.000000  119390.000000  ...        119390  103050.000000   
unique                   NaN            NaN  ...             3            NaN   
top                      NaN            NaN  ...    No Deposit            NaN   
freq                     NaN            NaN  ...        104641            NaN   
mean                2.500302       1.856403  ...           NaN      86.693382   
std                 1.908286       0.579261  ...           NaN     110.774548   
min                 0.000000       0.000000  ...           NaN       1.000000   
25%                 1.000000       2.000000  ...           NaN       9.000000   
50%                 2.000000       2.000000  ...           NaN      14.000000   
75%                 3.000000       2.000000  ...           NaN     229.000000   
max                50.000000      55.000000  ...           NaN     535.000000   

            company days_in_waiting_list customer_type            adr  \
count   6797.000000        119390.000000        119390  119390.000000   
unique          NaN                  NaN             4            NaN   
top             NaN                  NaN     Transient            NaN   
freq            NaN                  NaN         89613            NaN   
mean     189.266735             2.321149           NaN     101.831122   
std      131.655015            17.594721           NaN      50.535790   
min        6.000000             0.000000           NaN      -6.380000   
25%       62.000000             0.000000           NaN      69.290000   
50%      179.000000             0.000000           NaN      94.575000   
75%      270.000000             0.000000           

#### Unique values in each variables

In [101]:
for i in df.columns.tolist():
  print(f"Unique value for {i} : { df[i].nunique()}.")

Unique value for hotel : 2.
Unique value for is_canceled : 2.
Unique value for lead_time : 479.
Unique value for arrival_date_year : 3.
Unique value for arrival_date_month : 12.
Unique value for arrival_date_week_number : 53.
Unique value for arrival_date_day_of_month : 31.
Unique value for stays_in_weekend_nights : 17.
Unique value for stays_in_week_nights : 35.
Unique value for adults : 14.
Unique value for children : 5.
Unique value for babies : 5.
Unique value for meal : 5.
Unique value for country : 177.
Unique value for market_segment : 8.
Unique value for distribution_channel : 5.
Unique value for is_repeated_guest : 2.
Unique value for previous_cancellations : 15.
Unique value for previous_bookings_not_canceled : 73.
Unique value for reserved_room_type : 10.
Unique value for assigned_room_type : 12.
Unique value for booking_changes : 21.
Unique value for deposit_type : 3.
Unique value for agent : 333.
Unique value for company : 352.
Unique value for days_in_waiting_list : 128.


## ***3. Data Wrangling***

## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

## ***5. Solution to Business Objective***

## ***Conclusion***